modules/extensions-support/src/main/java/org/mule/module/extension/internal/manager/ExtensionEntityKey.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.manager;
+
+import org.mule.extension.annotation.api.Extension;
+
+final class ExtensionEntityKey
+{
+
+    private String name;
+    private String vendor;
+
+    protected ExtensionEntityKey(String name, String vendor)
+    {
+        if (vendor == null)
+        {
+            vendor = Extension.NO_VENDOR;
+        }
+        this.name = name;
+        this.vendor = vendor;
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (!(o instanceof ExtensionEntityKey))
```

Comment:
this is more clear to read if you remove the ! and the extra parenthesis and move the rest of the method inside the if block.

If the conditions doesn't checks out, you return false outside